mlir/include/mlir/IR/AttrTypeBase.td

```diff
@@ -283,6 +283,78 @@ class AttrDef<Dialect dialect, string name, list<Trait> traits = [],
                                  "::" # cppClassName # ">($_self)">;
 }
 
+// Define a StringAttr wrapper for the NamedAttribute `name`
```

Comment:
The "NamedAttribute" part isn't clear to me.

Also this is more than a "wrapper", it seems to me this is actually registering its own full fledge attribute? 
I'm not sure I follow the definition, but I suspect this won't be uniqued with StringAttr.

Reply:
Yes, this is no different than a StringAttr, so the usual interface works, but the new `get` only takes MLIRContext*. And once it is created, there is no way to get back to the wrapper class but it is mostly static lookup methods anyway.